modules/programs/nushell.nix

```diff
@@ -21,40 +21,55 @@ in {
       description = "The package to use for nushell.";
     };
 
-    settings = mkOption {
-      type = with types;
-        let
-          prim = oneOf [ bool int str ];
-          primOrPrimAttrs = either prim (attrsOf prim);
-          entry = either prim (listOf primOrPrimAttrs);
-          entryOrAttrsOf = t: either entry (attrsOf t);
-          entries = entryOrAttrsOf (entryOrAttrsOf entry);
-        in attrsOf entries // { description = "Nushell configuration"; };
-      default = { };
+    config = mkOption {
```

Comment:
Perhaps `configFile` and `envFile` to avoid the heavily overloaded `config` name?